spring-cloud-kubernetes-commons/src/main/java/org/springframework/cloud/kubernetes/commons/config/reload/ConfigReloadAutoConfiguration.java

```diff
@@ -64,22 +64,20 @@ public TaskSchedulerWrapper<TaskScheduler> taskScheduler() {
 	public ConfigurationUpdateStrategy configurationUpdateStrategy(ConfigReloadProperties properties,
 			ConfigurableApplicationContext ctx, Optional<RestartEndpoint> restarter, ContextRefresher refresher) {
 		String strategyName = properties.getStrategy().name();
-		switch (properties.getStrategy()) {
-		case RESTART_CONTEXT:
-			restarter.orElseThrow(() -> new AssertionError("Restart endpoint is not enabled"));
-			return new ConfigurationUpdateStrategy(strategyName, () -> {
-				wait(properties);
-				restarter.get().restart();
-			});
-		case REFRESH:
-			return new ConfigurationUpdateStrategy(strategyName, refresher::refresh);
-		case SHUTDOWN:
-			return new ConfigurationUpdateStrategy(strategyName, () -> {
+		return switch (properties.getStrategy()) {
+			case RESTART_CONTEXT -> {
+				restarter.orElseThrow(() -> new AssertionError("Restart endpoint is not enabled"));
+				yield new ConfigurationUpdateStrategy(strategyName, () -> {
+					wait(properties);
+					restarter.get().restart();
+				});
+			}
+			case REFRESH -> new ConfigurationUpdateStrategy(strategyName, refresher::refresh);
+			case SHUTDOWN -> new ConfigurationUpdateStrategy(strategyName, () -> {
 				wait(properties);
 				ctx.close();
 			});
-		}
-		throw new IllegalStateException("Unsupported configuration update strategy: " + strategyName);
```

Comment:
in general, you do not need a default case if all the enum values are covered. We throw here an `IllegalStateException` and say what `strategyName` to fail with. If we remove this line, we could, in theory, get an `IncompatibleClassChangeError` (that is the `default` that `javac` will add).

In our case, that is simply impossible, because `ReloadStrategy` is in the same maven module as this class, and you can't compile `ReloadStrategy` as `version-1`, then compile this class with the same `version-1`; only after to compile `ReloadStrategy` with `version-2` and and run this class. This simply makes no sense, as such drop this line over here